src/arkode/arkode_bandpre.c

```diff
@@ -478,27 +473,30 @@ static int ARKBandPDQJac(ARKBandPrecData pdata,
   int retval;
 
   ark_mem = (ARKodeMem) pdata->arkode_mem;
+  SUNAssignSUNCTX(ark_mem->sunctx);
 
   /* Access implicit RHS function */
   fi = NULL;
   fi = ark_mem->step_getimplicitrhs((void*) ark_mem);
   if (fi == NULL)  return(-1);
 
   /* Obtain pointers to the data for various vectors */
-  ewt_data   = N_VGetArrayPointer(ark_mem->ewt);
-  fy_data    = N_VGetArrayPointer(fy);
-  ftemp_data = N_VGetArrayPointer(ftemp);
-  y_data     = N_VGetArrayPointer(y);
-  ytemp_data = N_VGetArrayPointer(ytemp);
-  cns_data = (ark_mem->constraintsSet) ?
-    N_VGetArrayPointer(ark_mem->constraints) : NULL;
+  ewt_data   = SUNCheckCallLastErrNoRet(N_VGetArrayPointer(ark_mem->ewt));
+  fy_data    = SUNCheckCallLastErrNoRet(N_VGetArrayPointer(fy));
+  ftemp_data = SUNCheckCallLastErrNoRet(N_VGetArrayPointer(ftemp));
+  y_data     = SUNCheckCallLastErrNoRet(N_VGetArrayPointer(y));
+  ytemp_data = SUNCheckCallLastErrNoRet(N_VGetArrayPointer(ytemp));
+  cns_data   = NULL;
+  if (ark_mem->constraintsSet) {
+    cns_data = SUNCheckCallLastErrNoRet(N_VGetArrayPointer(ark_mem->constraints));
+  }
```

Comment:
Any of these errors should cause the program to halt:
```suggestion
  ewt_data   = SUNCheckCallLastErr(N_VGetArrayPointer(ark_mem->ewt));
  fy_data    = SUNCheckCallLastErr(N_VGetArrayPointer(fy));
  ftemp_data = SUNCheckCallLastErr(N_VGetArrayPointer(ftemp));
  y_data     = SUNCheckCallLastErr(N_VGetArrayPointer(y));
  ytemp_data = SUNCheckCallLastErr(N_VGetArrayPointer(ytemp));
  cns_data   = NULL;
  if (ark_mem->constraintsSet) {
    cns_data = SUNCheckCallLastErr(N_VGetArrayPointer(ark_mem->constraints));
  }
```